File: RecursionAndBacktracking/C++/print_increasing_decreasing.cpp
#include<bits/stdc++.h>
using namespace std;
void printIncreasing(int n) {
	if(n==0) return;
	printIncreasing(n-1);
	cout<<n<<'\n';
}

void printDecreasing(int n) {
	if(n==0) return;
	cout<<n<<'\n';
	printDecreasing(n-1);
}

int main() {
	int n;
	cin>>n;
	printIncreasing(n);
	cout<<"-----------------------------------"<<'\n';
	printDecreasing(n);

}

File: RecursionAndBacktracking/Java/recursion-backtracking/src/com/shubham/recursion/PrintIncreasingDecreasing.java
package com.shubham.recursion;

import java.util.Scanner;

public class PrintIncreasingDecreasing {
    static void printIncreasing(int n) {
        if(n==0) return;
        printIncreasing(n-1);
        System.out.println(n);
    }

    static void printDecreasing(int n) {
        if(n==0) return;
        System.out.println(n);
        printDecreasing(n-1);
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        printIncreasing(n);
        System.out.println("------------------------------------------");
        printDecreasing(n);
    }
}
